### PR TITLE
Adds python client examples

### PR DESCRIPTION
**Pull request by ecomba**
Adds examples for:

- list accounts
- auth
- bad auth
- check domains
- create contact
- create domain
- list domains
- get extended tld attributes
- list zone records

### CODE REVIEW

File: python/zone_records.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+In this case, `account` provides access to the contained account object.
+
+Here the account id is extracted for use in future calls:
+"""
+account_id = response.data.account.id
+
+"""
+dnsimple.client.zones.list_records is the method to list the records in a DNSimple zone.
+"""
+response = client.zones.list_records(account_id, str(sys.argv[1]))
```
Comment (onlyhavecans) [inline python/zone_records.py:40]: This only returns the first thirty records per our API. I recommend we demostrate iterating through pagination

**Comment by ecomba** [comment]
> The header comments for most files mention a "client token" and say an account token will work but not be useful. I think this is backward unless I am missing context. I think we should stick to User Token and Account token verbiage per our documentation https://developer.dnsimple.com/v2/#authentication

You are right! I took the headers from the Ruby examples. I will change them.

File: python/account_list.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+All calls to client pass through a service. In this case, `client.accounts` is the accounts service
+
+`client.accounts.list_accounts() is the method for retrieving the list of accounts the current authenticated
+entity has access to.
+"""
+response = client.accounts.list_accounts()
+
+account = response.data[0]
+
+print(f'Account= id:{account.id}, email:{account.email}, plan_identifier:{account.plan_identifier}')
```
[the remark this answers is not in the document]
Comment (ecomba) [reply python/account_list.py:26]: I took the headers from the Ruby examples and used them for the python examples.

You are right in that they might confuse the reader. I will update them.

[the remark this answers is not in the document]
Comment (ecomba) [reply python/account_list.py:26]: Does something like this sound better?

# This script assumes an account token.
#
# If all you have is a user token you can run this script, by passing email and password when creating the client
# like so:
# `client = Client(sandbox=True, email=<your_email>, password=<your_password>

File: python/zone_records.py (hunk 1 above)
Comment (ecomba) [reply python/zone_records.py:40]: It makes sense to me to show how pagination works to developers. I did not see any example of this in the examples from the other clients, so it did not occur to me to show this here.

File: python/account_list.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (onlyhavecans) [reply python/account_list.py:26]: That looks great

**Review by onlyhavecans — CHANGES_REQUESTED** [review]
I found a bug in the examples when running them.
None of these examples account for pagination. I have a test domain with 10_000 test records but if I run `./zone_records.py witchcraft.test | wc -l` I get 30, our default page size

Below is a recommendation on how we could demonstrate pagination. I am not sure if this is the most optimal way but was my spike at it. 

I really think we should show pagination in the examples as it can and does trip up new developers if  they skim the docs.

```python
page = 1
while True:
    response = client.zones.list_records(account_id, str(sys.argv[1]), page=page)

    for record in response.data:
        print(f'- {record.content} ({record.id})')

    if page == response.pagination.total_pages:
        break

    page += 1
```

**Comment by ecomba** [comment]
> I found a bug in the examples when running them.
> None of these examples account for pagination. I have a test domain with >200 test records but if I run `./zone_records.py witchcraft.test | wc -l` I get 30, our default page size
> 
> Below is a recommendation on how we could demonstrate pagination. I am not sure if this is the most optimal way but was my spike at it.
> 
> I really think we should show pagination in the examples as it can and does trip up new developers if they skim the docs.
> 
> ```python
> page = 1
> while True:
>     response = client.zones.list_records(account_id, str(sys.argv[1]), page=page)
> 
>     for record in response.data:
>         print(f'- {record.content} ({record.id})')
> 
>     if page == response.pagination.total_pages:
>         break
> 
>     page += 1
> ```

Is this a bug? Or just that we are not paginating through all the pages in the examples?

I will add the pagination to the example 😄

File: python/zone_records.py (hunk 1 above)
Comment (ecomba) [reply python/zone_records.py]: I have now added this to the `zone_records.py` example.

**Review by onlyhavecans — APPROVED** [review]
👍

**Pull request merged**